### ISSUE TITLE
Ultisnips broken after sierra install

### ISSUE THREAD

**Issue by chrisrytting** [SirVer/ultisnips#823]
When I upgraded to Mac OSX Sierra, Ultisnips broke for me. I have tried everything from wiping the files on my machine to following instructions from other issues here to using a different package manager and nothing seems to be doing the trick. I get a whole slew of errors when I try to do basically anything in vim when ultisnips is active, like unto the problem addressed in Bug #1156584 reported by Chris Clarke on 2013-03-18 on https://bugs.launchpad.net/ultisnips/+bug/1156584. What was suggested there didn't do the trick for me though. 

Machine specs:
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jun 10 2015 11:31:44)
MacOS X (unix) version
Included patches: 1-712
Compiled by Homebrew
Huge version with MacVim GUI. 

First error I get after installing ultisnips is this:

`Error detected while processing /Users/chrisrytting1/.vim/bundle/ultisnips/autoload/UltiSnips.vim:
line    8:
Traceback (most recent call last):`

and the following happens after pressing a few buttons trying to type.

`Error detected while processing function UltiSnips#TrackChange:
line    1:
Traceback (most recent call last):
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
  File "<string>", line 1, in <module>
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
NameError: name 'UltiSnips_Manager' is not defined
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
Traceback (most recent call last):
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
  File "<string>", line 1, in <module>
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
NameError: name 'UltiSnips_Manager' is not defined
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
Traceback (most recent call last):
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
  File "<string>", line 1, in <module>
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
NameError: name 'UltiSnips_Manager' is not defined
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
Traceback (most recent call last):
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:
  File "<string>", line 1, in <module>
Press ENTER or type command to continue
Error detected while processing function UltiSnips#TrackChange:
line    1:`

Let me know what else I can tell you but please help me fix this. I love ultisnips and want to come back to the fold.

**Comment by seletskiy** [comment]
@chrisrytting: Disable autotrigger feature by adding `au! UltiSnips_AutoTrigger` in your configuration. Why it's causing mentioned import error is need to be investigated, I do not have Mac and issue doesn't reproduce under Linux.

**Comment by Cheezmeister** [comment]
Experiencing this too. A bunch of error prompts occur when pressing any key in insert mode. (Entering insert mode, then immediately exiting, does not trigger it.)

This renders the plugin not only useless but actively hostile on OSX. It's a real shame.

```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Feb  4 2017 00:48:24)
Included patches: 1-898, 8056
Compiled by root@apple.com
Normal version without GUI.  Features included (+) or not (-):
+acl             +cindent         -conceal         +digraphs        -farsi           -gettext         +libcall         +mksession       -mouse_netterm   -mzscheme        -profile         +scrollbind      +syntax          +textobjects     +visualextra     -X11
-arabic          -clientserver    +cryptv          -dnd             +file_in_path    -hangul_input    +linebreak       +modify_fname    -mouse_sgr       +netbeans_intg   +python/dyn      +signs           +tag_binary      +title           +viminfo         -xfontset
+autocmd         -clipboard       +cscope          -ebcdic          +find_in_path    +iconv           +lispindent      +mouse           -mouse_sysmouse  +path_extra      -python3         +smartindent     +tag_old_static  -toolbar         +vreplace        -xim
-balloon_eval    +cmdline_compl   +cursorbind      -emacs_tags      +float           +insert_expand   +listcmds        -mouseshape      -mouse_urxvt     -perl            +quickfix        -sniff           -tag_any_white   +user_commands   +wildignore      -xsmp
-browse          +cmdline_hist    +cursorshape     +eval            +folding         +jumplist        +localmap        -mouse_dec       +mouse_xterm     +persistent_undo +reltime         +startuptime     -tcl             +vertsplit       +wildmenu        -xterm_clipboard
+builtin_terms   +cmdline_info    +dialog_con      +ex_extra        -footer          -keymap          -lua             -mouse_gpm       +multi_byte      +postscript      -rightleft       +statusline      +terminfo        +virtualedit     +windows         -xterm_save
+byte_offset     +comments        +diff            +extra_search    +fork()          -langmap         +menu            -mouse_jsbterm   +multi_lang      +printer         +ruby/dyn        -sun_workshop    +termresponse    +visual          +writebackup     -xpm
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L/usr/local/lib -o vim        -lm -lncurses  -liconv
```

Can confirm @seletskiy's fix works for me.

**Comment by iSE7EN** [comment]
I'm use macOS Sierra Version 10.12.5

fix error with vim 8 from macport

$`sudo port install vim +python27` 
$`port echo installed|grep vim`

> vim                            @8.0.0596_0+huge 
> vim                            @8.0.0596_0+huge+python27
> 

$`sudo port activate vim @8.0.0596_0+huge+python27`

**Comment by dopey** [comment]
Same Problem

Machine specs:
macOS Sierra 10.12.5
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Nov  5 2016 21:51:05)
MacOS X (unix) version
Included patches: 1-66
Compiled by Homebrew

When I try to set `au! UltiSnips_AutoTrigger` I get: `E216: No such group or event: UltiSnips_AutoTrigger`

**Comment by blazzy** [comment]
I've been running into this issue with the stock vim on Sierra:
`VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Feb  4 2017 00:48:24)`

My MacVim install has been fine though:
`VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Sep 13 2017 04:07:30)`

I updated my CLI vim with `brew` and that version works fine too:
`VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Sep 14 2017 18:28:18)`

**Comment by denzelwamburu** [comment]
Am still having the same issues.

**Comment by SirVer** [comment]
See the fixes in #771, my explanation is in https://github.com/SirVer/ultisnips/issues/771#issuecomment-377815616.
